napari/layers/_layer_actions.py

```diff
@@ -105,12 +105,47 @@ def _convert_to_image(ll: LayerList) -> None:
 def _merge_stack(ll: LayerList, rgb: bool = False) -> None:
     # force selection to follow LayerList ordering
     imgs = cast(list[Image], [layer for layer in ll if layer in ll.selection])
-    assert all(isinstance(layer, Image) for layer in imgs)
-    merged = (
-        stack_utils.merge_rgb(imgs)
-        if rgb
-        else stack_utils.images_to_stack(imgs)
+    assert all(isinstance(layer, Image) for layer in imgs), trans._(
```

Comment:
`assert` should be used only in tests. In production code, we should raise `ValueError`

Reply:
Can you explain why @Czaki?

Reply:
Because assert could be disabled using flag `-O`